nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,66 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    const mealId = parseInt(request.params.id);
+    if (isNaN(mealId)) {
+      response.status(400).json("id must be an integer");
+      return;
+    } if (mealId <= meals.length) {
+      const mealById = meals.filter((meal) => meal.id === mealId);
+      console.log(mealById);
+      response.json(mealById[0]);
+    } else {
+      response.json({});
+    }
+  } catch (error) {
+    throw error;
```

Comment:
Throwing errors that are not caught would crash the server. Instead you could return an server error code:

```suggestion
    response.sendStatus(500);
```